rero_ils/modules/items/api/issue.py

```diff
@@ -47,3 +52,144 @@ def issue_is_regular(self):
+
+    @classmethod
+    def get_late_serial_holdings_pids(cls):
+        """Return pids for all holdings of type serial and next expected date.
+
+        is greater than sysdate.
+
+        :return a generator of holdings pid
+        """
+        from ...holdings.api import HoldingsSearch
+        current_date = datetime.now(timezone.utc).strftime('%Y-%m-%d')
+        query = HoldingsSearch() \
+            .filter('term', holdings_type='serial') \
+            .filter('range',
+                    patterns__next_expected_date={'lte': current_date})
+        results = query\
+            .params(preserve_order=True) \
+            .sort({'_created': {'order': 'asc'}}) \
+            .source(['pid']).scan()
+        for hit in results:
+            yield hit.pid
+
+    @classmethod
+    def receive_next_late_expected_issues(cls, dbcommit=False, reindex=False):
+        """Receive the next late expected issue for all holdings."""
+        from ...holdings.api import Holding
+        created_issues = 0
+        pids = cls.get_late_serial_holdings_pids()
+        for pid in pids:
+            holding = Holding.get_record_by_pid(pid)
+            issue = holding.receive_regular_issue(
+                dbcommit=dbcommit, reindex=reindex)
+            issue['issue']['status'] = ItemIssueStatus.LATE
+            issue = issue.update(issue, dbcommit=dbcommit, reindex=reindex)
+            created_issues += 1
+        return created_issues
+
+    @classmethod
+    def create_first_claim(cls, dbcommit=False, reindex=False):
+        """Claim the late issues for all holdings."""
+        from ...holdings.api import Holding
+        modified_issues = 0
+        issues = cls.get_issues_by_status(issue_status=ItemIssueStatus.LATE)
+        for issue in issues:
+            email = None
+            holding = Holding.get_record_by_pid(issue.holding_pid)
+            vendor = holding.vendor
+            if vendor:
+                email = vendor.order_email
+            max_number_of_claims = holding.max_number_of_claims
+            if email and max_number_of_claims and \
+                    max_number_of_claims > issue.claims_count:
+                if issue.claims_count == 0 and holding.days_before_first_claim:
+                    issue_status_date = ciso8601.parse_datetime(
+                        issue.issue_status_date)
+                    days = (
+                        datetime.now(timezone.utc) - issue_status_date).days
+                    if days >= holding.days_before_first_claim:
+                        issue['issue']['status'] = ItemIssueStatus.CLAIMED
+                        issue['issue']['claims_count'] = issue.claims_count + 1
+                        issue = issue.update(
+                            issue, dbcommit=dbcommit, reindex=reindex)
+                        modified_issues += 1
+        return modified_issues
+
+    @classmethod
+    def create_next_claims(cls, dbcommit=False, reindex=False):
```

Comment:
This method seems similar that the previous one. You can consider merge them into on method if it is possible.